AI-AGENTS.md

```diff
@@ -0,0 +1,142 @@
+# AI Coding Agent Guide
+
+This file contains project-specific instructions that any AI coding agent should read at the start of each conversation and maintain in memory throughout the entire interaction. IMPORTANT: Once this file has been read or updated, it MUST be loaded at the beginning of any new conversation to ensure awareness of communication requirements, custom tasks, etc.
+
+## Default Mode
+
+- Architect mode should be enabled by default.
+- Focus on providing detailed analysis, patterns, trade-offs, and architectural guidance.
+- If you're unable to write code that fits these guidelines, stop and ask for additional context from the developer.
+
+## Development Environment
+
+Discourse is a large project with a long development history. Ensure you understand the context of any changes you're making before you start.
+
+### General Rules
+These rules apply to ALL files being changed.
+
+- Always lint changed files.
+- Always make display strings translatable.
+- Avoid splitting display strings into pieces: use translation placeholders, or multiple translatable strings where appropriate.
+- After completing a task, create a subagent to review the changes and ensure they conform to the instructions in this file, as well as the prompt(s) given.
+
+### JavaScript
+- Don't create empty backing classes for template tag only components, unless specifically asked to.
+- Use the FormKit library for creating forms and form inputs. FormKit is documented here: https://meta.discourse.org/t/discourse-toolkit-to-render-forms/326439
+
+### JavaScript Documentation
+- Always add JSDocs for classes, methods, and members, except for:
+  - `@service` members
+  - constructors
+- Always use multiline JSDoc format.
+- For components:
+  - Specify the component name with `@component`.
+  - List the params. These can be found in `this.args` in the JS, or `@paramname` in the `<template>`.
+- For methods:
+  - Don't add `@returns` for `@action` methods.
+  - Don't add `@type` for getters, document with `@returns`.
+- For members:
+  - Specify the @type.
```

Comment:
I'm not entirely sure if all of these are correct, do we have defined JSDoc standards?